codenvy-ide-core/src/test/java/com/codenvy/ide/part/editor/EditorPartStackPresenterTest.java

```diff
@@ -0,0 +1,355 @@
+        onSuccess2.invoke(callback2, (Void)null);
+
+        verify(view).removeTab(anyInt());
+
+        assertTrue(presenter.getNumberOfParts() == 1);
+
+        verify(part2).removePropertyListener(any(PropertyListener.class));
+
+        verify(partStackEventHandler, never()).onActivePartChanged(part2);
+
+        //check closing third tab
+        reset(view);
+        reset(partStackEventHandler);
+        ArgumentCaptor<AsyncCallback> asyncRequestCallbackCaptor3 = ArgumentCaptor.forClass(AsyncCallback.class);
+
+        verify(part3).onClose(asyncRequestCallbackCaptor3.capture());
+
+        AsyncCallback callback3 = asyncRequestCallbackCaptor3.getValue();
+
+        Method onSuccess3 = GwtReflectionUtils.getMethod(callback3.getClass(), "onSuccess");
+        onSuccess3.invoke(callback3, (Void)null);
+
+        verify(view).removeTab(anyInt());
+
+        assertTrue(presenter.getNumberOfParts() == 0);
+
+        verify(part3).removePropertyListener(any(PropertyListener.class));
+
+        verify(partStackEventHandler).onActivePartChanged(any(EditorPartPresenter.class));
+
+        verify(asyncCallback, never()).onSuccess(null);
+    }
+
+    @Test
+    public void setNewActivePartTest() {
+        doReturn(item).when(view).addTabButton(null, null, null, null, true);
+        presenter.addPart(part1);
+        stopSchedulers();
+
+        presenter.setActivePart(part1);
+
+        TestCase.assertEquals(presenter.getActivePart(), part1);
+
+        verify(view).setActiveTab(0);
+
+        assertEquals(presenter.getActivePart(), part1);
+    }
+
+    @Test
+    public void addFirstPartWithoutIconTest() {
+        doReturn(item).when(view).addTabButton(null, null, null, null, true);
+
+        int amountOfPartsBefore = presenter.getNumberOfParts();
+
+        presenter.addPart(part1);
+        stopSchedulers();
+
+        int amountOfPartsAfter = presenter.getNumberOfParts();
+
+        assertTrue(++amountOfPartsBefore == amountOfPartsAfter);
+
+        verify(part1).addPropertyListener(any(PropertyListener.class));
+
+        verify(part1).getTitleSVGImage();
+
+        verify(view).addTabButton(null, null, null, null, true);
+
+        verify(item).addClickHandler(any(ClickHandler.class));
+        verify(item).addCloseHandler(any(CloseHandler.class));
+
+        verify(part1).go(any(AcceptsOneWidget.class));
+
+        assertEquals(presenter.getActivePart(), part1);
+        //because this is first tab
+        verify(view).setActiveTab(eq(0));
+
+        verify(partStackEventHandler).onRequestFocus(eq(presenter));
+    }
+
+    @Test
+    public void addFirstPartWithIconTest() {
+        SVGResource titleSVGResource = mock(SVGResource.class);
+        SVGImage svgImage = mock(SVGImage.class);
+        OMSVGSVGElement svgElem = mock(OMSVGSVGElement.class);
+
+        doReturn(svgElem).when(titleSVGResource).getSvg();
+        doReturn(GWT.create(Node.class)).when(svgElem).getElement();
```

Comment:
Yes, I need GWT class, because mock faild in native javascript code which situated inside in getElement() method. It's difficult testcase.